src/main/scala/edu/berkeley/ce/rockslicing/RockSlicer.scala

```diff
@@ -47,14 +50,65 @@ object RockSlicer {
       Block(center, block.nonRedundantFaces)
     }
 
-    // Calculate centroid of each block
-    val centroidBlocks = nonRedundantBlocks.map { block =>
+    // Find all blocks that contain processor joints
+    val processorBlocks = nonRedundantBlocks.filter { block => 
+      block.faces.exists { face => face.processorJoint}
+    }
+
+    // Find blocks that do not contain processor joints
+    val realBlocks = nonRedundantBlocks.filter { block =>
+      val faceFlags = block.faces map { face =>
+        face.processorJoint
+      }
+      !faceFlags.contains(true)
+    }
+
+    // PROCESS PROCESSOR BLOCKS BEFORE CONTINUEING WITH REMAINING BLOCKS
+    // Collect all blocks that contain processor joints from all nodes
+    val allProcessorBlocks = processorBlocks.collect()
+    // Search blocks for matching processor joints
+    if (allProcessorBlocks.length > 0) {
+      val reconstructedBlocks = mergeBlocks(allProcessorBlocks, Seq[Block](), globalOrigin)
+
+      // Update centroids of reconstructed processor blocks and remove duplicates
+      val reconCentroidBlocks = reconstructedBlocks.map {block =>
+        val centroid = block.centroid
+        Block(centroid, block.updateFaces(centroid))
+      }
+
+      // Clean up faces of reconstructed blocks with values that should be zero, but have
+      // arbitrarily small floating point values
+      val squeakyCleanRecon = reconCentroidBlocks.map { case Block(center, faces) =>
+        Block(center, faces.map(_.applyTolerance))
+      }
+
+      // Convert reconstructed blocks to requested output
+      if (arguments.toInequalities) {
+        val jsonReconBlocks = squeakyCleanRecon.map(Json.blockToMinimalJson)
```

Comment:
The reason I separated them into their own file is because all the "processor" have been collected onto one node. Spark's text file utilities don't work in this instance. 
I was thinking about how we can have all the nodes participate in this step instead of having only one node do all this work. If we do that, we can use Spark's text file utilities. I think grouping the blocks based on them containing the same processor joint and then parallelizing the collection to merge them will be a better approach. Currently looking into this.